core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -743,5 +743,22 @@ trait EntityComponent {
         entityRec.id -> unmarshalEntity(entityRec, attributesByEntityId.getOrElse(entityRec.id, Map.empty))
       }
     }
+
+    object EntityStatisticsQueries extends RawSqlQuery {
+      val driver: JdbcDriver = EntityComponent.this.driver
+      def countEntitiesofTypeInNamespace(workspaceNamespace: Option[String], workspaceName: Option[String]): ReadAction[Seq[NamedStatistic]] = {
+        val baseSelect = sql"""select entity_type, count(1) as entityCount
+                from ENTITY e, WORKSPACE w
+                where e.workspace_id = w.id
+                  and e.deleted = 0"""
+        val namespaceClause = workspaceNamespace.map(ns => sql" and w.namespace = $ns").getOrElse(sql"")
+        val nameClause = workspaceName.map(n => sql" and w.name = $n").getOrElse(sql"")
```

Comment:
I can't tell if `workspaceNamespace` and `workspaceName` are protected against sql injection here. It doesn't look like `concatSqlActions` does anything to the string values in the name clauses. Is this handled by the `sql` macro?

Reply:
yeah, the `sql` macro turns all variables into bind variables to protect against this

see http://slick.lightbend.com/doc/3.0.0/sql.html#string-interpolation

Reply:
👍  Gotcha, thanks!